agent/src/softwarecontaineragent.cpp

```diff
@@ -316,106 +338,88 @@ bool SoftwareContainerAgent::execute(ContainerID containerID,
 
     // Start it
     if (isError(job->start())) {
-        log_error() << "Could not start job";
-        pid = INVALID_PID;
-        return false;
+        std::string errorMessage("Could not start job");
+        log_error() << errorMessage;
+        throw SoftwareContainerError(errorMessage);
     }
 
-    pid = job->pid();
     // If things went well, do what we need when it exits
     addProcessListener(m_connections, job->pid(), [listener](pid_t pid, int exitCode) {
         listener(pid, exitCode);
     }, m_mainLoopContext);
 
     profilepoint("executeEnd");
 
-    return true;
+    return job->pid();
 }
 
-bool SoftwareContainerAgent::shutdownContainer(ContainerID containerID)
+void SoftwareContainerAgent::shutdownContainer(ContainerID containerID)
 {
     profilefunction("shutdownContainerFunction");
     if (!m_shutdownContainers) {
-        log_info() << "Not shutting down containers, so not shutting down " << containerID;
-        return false;
+        std::string errorMessage("Not shutting down containers, so not shutting down ID: " + std::to_string(containerID));
+        log_error() << errorMessage;
+        throw SoftwareContainerError(errorMessage);
```

Comment:
This should not be an error imo, this is just a fact. Just returning would suffice here.